### ISSUE TITLE
Improve integration tests.

### ISSUE THREAD

**Issue by sunshinejr** [Moya/Moya#1424]
Right now few of the integration tests we have (look at the tests failed [here](https://circleci.com/gh/Moya/Moya/2211?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)) are failing whenever the internet connection is worse than normal. The simplest fix would be to increase the timer from 5 seconds to e.g. 10. But, if you have _not-that-great_ internet connection and you still want to run tests after your changes, it might be difficult with our current setup.

Thus, the idea would be to create a separate target for integration tests, that you would run whenever you feel comfortable, and then you could fire up unit tests each time you change something in your development process.

## What's good to know before doing this task?
- Our integration tests are spread in multiple files (not only with  so you would have to search for them and move around
- Some of our tests are failing whenever our CI have worse internet connection for a while (see e.g. [this build](https://circleci.com/gh/Moya/Moya/2211?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link))

## What are the steps to implement this task?
- Create new target/scheme for integration tests
- Move tests to the new files/target
- Make sure all our tests are passing
- Update CI scripts (circle.yml & Rakefile)

Go ahead and modify this issue if you I forgot about something 😉

**Comment by BasThomas** [comment]
Yes, I fully support this change. (Having to make) a more conscious decision of the type of test you’re writing never hurts either. 💯 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
This seems to be a more pressing issue as of lately. I can never get builds to pass on CI because of these tests.

**Comment by sunshinejr** [comment]
@SD10 we can try to make a workaround with increased timeouts on these tests until we can improve them later on. Want to take a stab at it?

**Comment by SD10** [comment]
@sunshinejr Sure, what do you think is a reasonable timeout interval? I'm thinking 10 seconds. I guess I can experiment until it works again 😓

**Comment by sunshinejr** [comment]
Yeah 10 secs should be enough, if not maybe there is another problem in these two 🤔

**Comment by SD10** [comment]
@sunshinejr Think we can comment out the failing test? There are 6. I'm not sure if there is a deeper problem but sometimes only 3 fail locally for me. So I think it's a network problem. We need to get CI green again.

**Comment by sunshinejr** [comment]
Hmm, I would rather get it fixed or get a clear understanding of what is happening. Maybe we broke something in the process and just missed it because these tests were failing quite often. I can't really check this until later this week, but maybe someone from @Moya/contributors is available and ready for some digging?
